store/tikv/scan.go

```diff
@@ -199,7 +202,7 @@ func (s *Scanner) getData(bo *Backoffer) error {
 			// No more data in current Region. Next getData() starts
 			// from current Region's endKey.
 			s.nextStartKey = loc.EndKey
-			if len(loc.EndKey) == 0 {
+			if len(loc.EndKey) == 0 || (len(s.endKey) > 0 && kv.Key(s.nextStartKey).Cmp(kv.Key(s.endKey)) >= 0) {
```

Comment:
I find `s.endKey == nil` and `s.endKey == []byte{}` is not distinguished
Should we make it clear? or it's deliberately?

s.endKey == nil          -- nobody set it
s.endKey == []byte{}  -- it's actually set to a wrong value